tests/components/shelly/test_cover.py

```diff
@@ -91,3 +91,101 @@ async def test_no_roller_blocks(hass, coap_wrapper, monkeypatch):
     )
     await hass.async_block_till_done()
     assert hass.states.get("cover.test_name") is None
+
+
+async def test_rpc_device_services(hass, rpc_wrapper, monkeypatch):
+    """Test RPC device cover services."""
+    assert rpc_wrapper
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(rpc_wrapper.entry, COVER_DOMAIN)
+    )
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_SET_COVER_POSITION,
+        {ATTR_ENTITY_ID: "cover.test_cover_0", ATTR_POSITION: 50},
+        blocking=True,
+    )
+    state = hass.states.get("cover.test_cover_0")
+    assert state.attributes[ATTR_CURRENT_POSITION] == 50
+
+    monkeypatch.setitem(rpc_wrapper.device.status["cover:0"], "state", "opening")
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_OPEN_COVER,
+        {ATTR_ENTITY_ID: "cover.test_cover_0"},
+        blocking=True,
+    )
+    rpc_wrapper.async_set_updated_data("")
+    assert hass.states.get("cover.test_cover_0").state == STATE_OPENING
+
+    monkeypatch.setitem(rpc_wrapper.device.status["cover:0"], "state", "closing")
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_CLOSE_COVER,
+        {ATTR_ENTITY_ID: "cover.test_cover_0"},
+        blocking=True,
+    )
+    rpc_wrapper.async_set_updated_data("")
+    assert hass.states.get("cover.test_cover_0").state == STATE_CLOSING
+
+    monkeypatch.setitem(rpc_wrapper.device.status["cover:0"], "state", "closed")
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_STOP_COVER,
+        {ATTR_ENTITY_ID: "cover.test_cover_0"},
+        blocking=True,
+    )
+    rpc_wrapper.async_set_updated_data("")
+    assert hass.states.get("cover.test_cover_0").state == STATE_CLOSED
+
+
+async def test_rpc_device_no_cover_keys(hass, rpc_wrapper, monkeypatch):
+    """Test RPC device without cover keys."""
+    assert rpc_wrapper
+
+    monkeypatch.delitem(rpc_wrapper.device.status, "cover:0")
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(rpc_wrapper.entry, COVER_DOMAIN)
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get("cover.test_cover_0") is None
+
+
+async def test_rpc_device_update(hass, rpc_wrapper, monkeypatch):
+    """Test RPC device update."""
+    assert rpc_wrapper
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(rpc_wrapper.entry, COVER_DOMAIN)
+    )
+    await hass.async_block_till_done()
+
+    monkeypatch.setitem(rpc_wrapper.device.status["cover:0"], "state", "closed")
+    await hass.helpers.entity_component.async_update_entity("cover.test_cover_0")
```

Comment:
Please move time forward by firing a time changed event instead of using `async_update_entity`, if this is a polling entity.

https://github.com/home-assistant/core/blob/2791cd68c2eba1a926eb4c6b0c8d38076330343b/tests/common.py#L380-L384

Side note: Using `hass.helpers` instead of importing names as normal is legacy.

Reply:
This is actually not a polling integration, the basic mock we use is not optimal so we don't really push updates when mocking. For other integrations I already wrote a better mock which allow mocking the library push updates.

It is on my task list to rewrite the tests for Shelly and also complete the coverage to 100%, the problem is that Shelly keeps making new products and we have hard time catching up with them 😃.

Since the cover platform is small, I will try to convert it to a better mock in few weeks from now.

Reply:
Ok. Then for now please just change it to import `async_update_entity` instead of using `hass.helpers`.